### PR TITLE
Update concepts and add terminology page

### LINKED ISSUE #25: Write terminology conceptual doc
https://github.com/dapr/docs/blob/master/concepts/terminology/terminology.md

### LINKED ISSUE #503: Dapr general architecture doc
Add in the architecture section, documentation which explains in detail the architecture of Dapr. different system services, side car etc and how they relate to each other.
This doc should not be Kubernetes specific but should link to hosting environments docs which details how this architecture is implemented in different hosting envs.

### LINKED ISSUE #1022: Need conceptual doc on control plane
Need a document on control plane in concepts

### LINKED ISSUE #1048: Comments on the Dapr overview
**Describe the issue**
You need to change https://v1-rc2.docs.dapr.io/concepts/overview/ in 2 ways, shown below.

One -- In the “Developer languages SDKs and frameworks.Web” section at https://v1-rc2.docs.dapr.io/concepts/overview/#web
It now says “In the Dapr .NET SDK you can find ASP.NET Core integration, which brings stateful routing controllers that respond to pub/sub events from other services.”

This **also needs to mention ASP.NET Core gRPC Services** that can use all Dapr features as well.

Two -- In the “Run Anywhere.Running Dapr on a local developer machine in self hosted  mode” diagram at https://v1-rc2.docs.dapr.io/concepts/overview/#running-dapr-on-a-local-developer-machine-in-self-hosted-mode.

In this diagram the double-ended arrow between the service code of the app and the Dapr Sidecar is labeled as follows:
**Dapr Api
Tracking, logs and metrics**

This would be clearer that there are 2 kinds of traffic over the double-ended arrow if labled as follows since there is plenty of space
**Dapr Api requests/responses
Tracking, logs and metrics**

**URL of the docs**
https://v1-rc2.docs.dapr.io/concepts/overview/
https://v1-rc2.docs.dapr.io/concepts/overview/#web
https://v1-rc2.docs.dapr.io/concepts/overview/#running-dapr-on-a-local-developer-machine-in-self-hosted-mode

**Expected content**
Please see the "Describe the issue" above.
**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by AaronCrawfis**
Closed #25

### CODE REVIEW

File: daprdocs/content/en/concepts/terminology.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+|:-----|------------|------------------|
+| App/Application | A running service/binary, usually that you as the user create and run.
+| Building block | A piece of functionality that Dapr provides to users to help in the creation of microservices and applications. | [Dapr building blocks]({{< ref building-blocks-concept.md >}})
+| Component | A modular piece of functionality that make up, either by itself or with a collection of other components, a Dapr building block. | [Dapr components]({{< ref components-concept.md >}})
+| Configuration | A YAML file declaring all of the settings for Dapr sidecars and the Dapr control plane. It is here where you can configure mTLS, tracing, and middleware. | [Dapr configuration]({{< ref configuration-concept.md >}})
+| Dapr | Distributed Application Runtime. | [Dapr overview]({{< ref overview.md >}})
+| Self-hosted | Your local Windows/macOS/Linux machine where you run your applications with Dapr. Dapr provides capabilities to run on your machine in "self-hosted" mode. | [Self-hosted mode]({{< ref self-hosted-overview.md >}})
+| Service | A running application or binary. Can be used to refer to your application, or a Dapr applicaiton.
```
Comment (orizohar) [inline daprdocs/content/en/concepts/terminology.md]: typo in the word "application"

File: daprdocs/content/en/concepts/terminology.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+| Sidecar | A program that runs along side your application as a separate binary or pod. | [Sidecar pattern](https://docs.microsoft.com/en-us/azure/architecture/patterns/sidecar)
+| System services | Applications that run as part of the Dapr control plane, which take care of the various jobs that Dapr handles such as service discovery or sidecar injection. | [Self-hosted overview]({{< ref self-hosted-overview >}})<br />[Kubernetes overview]({{< ref kubernetes-overview >}})
```
Comment (orizohar) [inline daprdocs/content/en/concepts/terminology.md]: "system services" is too broad, I don't think we only use that to describe the Dapr control plane.
Perhaps define the term "Dapr control plane" and in that case I would say it is "A collection of services that are part of a Dapr installation on a hosting platform such as a Kubernetes cluster, and allow Dapr enabled applications to run on that platform"

File: daprdocs/content/en/concepts/components-concept.md [hunk 3]
```diff
@@ -14,18 +14,34 @@ Dapr uses a modular design where functionality is delivered as a component. Each
 
  The following are the component types provided by Dapr:
 
-* [Bindings](https://github.com/dapr/components-contrib/tree/master/bindings)
-* [Pub/sub](https://github.com/dapr/components-contrib/tree/master/pubsub)
-* [Middleware](https://github.com/dapr/components-contrib/tree/master/middleware)
+* [State stores](https://github.com/dapr/components-contrib/tree/master/state)
```
Comment (orizohar) [inline daprdocs/content/en/concepts/components-concept.md]: While we're at it, should these just all link to the relevant component reference section in the docs?

File: daprdocs/content/en/concepts/terminology.md (hunk 1 above)
Comment (orizohar) [reply daprdocs/content/en/concepts/terminology.md]:
```suggestion
| Service | A running application or binary. Can be used to refer to your application, or a Dapr application.
```

File: daprdocs/content/en/concepts/terminology.md (hunk 2 above)
Comment (orizohar) [inline daprdocs/content/en/concepts/terminology.md]:
```suggestion
| Dapr control plane | A collection of services that are part of a Dapr installation on a hosting platform such as a Kubernetes cluster, and allow Dapr enabled applications to run on that platform. | [Self-hosted overview]({{< ref self-hosted-overview >}})<br />[Kubernetes overview]({{< ref kubernetes-overview >}})
```

File: daprdocs/content/en/concepts/components-concept.md (hunk 3 above)
Comment (AaronCrawfis) [reply daprdocs/content/en/concepts/components-concept.md]: Good idea. Updated

**Comment by orizohar** [comment]
@msfussell can you please review?

File: daprdocs/content/en/concepts/overview.md [hunk 4]
```diff
@@ -43,84 +43,78 @@ Each of these building blocks is independent, meaning that you can use one, some
...
 
-## Hosting Environments
-Dapr can be hosted in multiple environments, including self hosted for local development or to deploy to a group of VMs, Kubernetes and edge environments such as Azure IoT Edge.
+<img src="/images/overview-sidecar-model.png" width=700>
 
-### Self hosted
+## Hosting environments
 
-In self hosted mode Dapr runs as a separate side-car process which your service code can call via HTTP or gRPC. In self hosted mode, you can  also deploy Dapr onto a set of VMs.
+Dapr can be hosted in multiple environments, including self-hosted for local development and Kubernetes for production workloads.
```
Comment (msfussell) [inline daprdocs/content/en/concepts/overview.md]: Self-hosted is not just about local development. With the introduction of Consul, the VM scenario is now possible. You may not want to change this here, but it should be made clear about VMs being a hosting option.

File: daprdocs/content/en/concepts/overview.md [hunk 5]
```diff
@@ -43,84 +43,78 @@ Each of these building blocks is independent, meaning that you can use one, some
...
+| [Python]({{< ref python >}}) | [Flask]({{< ref python-flask.md >}})
+| [Javascript](https://github.com/dapr/js-sdk) | [Express](http://expressjs.com/)
+| [PHP]({{< ref php >}}) | | You can serve with Apache, Nginx, or Caddyserver.
 
-<img src="/images/overview_standalone.png" width=800>
+#### Integrations
 
-### Running Dapr in Kubernetes mode
+Visit the [integrations]({{< ref integrations >}}) page to learn about some of the first-class support Dapr has for various frameworks and external products, including:
+- Visual Studio Code
```
Comment (msfussell) [inline daprdocs/content/en/concepts/overview.md]: This is not really an integration VSCode. I would remove. And if we are going to have a list here (which may get a bit long) we need to add Azure API Management (APIM) as one of the more useful integrations. I would add this for now and we need to address the issue for this at some point

File: daprdocs/content/en/concepts/terminology.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+| Component | A modular piece of functionality that make up, either by itself or with a collection of other components, a Dapr building block. | [Dapr components]({{< ref components-concept.md >}})
+| Configuration | A YAML file declaring all of the settings for Dapr sidecars and the Dapr control plane. It is here where you can configure mTLS, tracing, and middleware. | [Dapr configuration]({{< ref configuration-concept.md >}})
+| Dapr | Distributed Application Runtime. | [Dapr overview]({{< ref overview.md >}})
+| Dapr control plane | A collection of services that are part of a Dapr installation on a hosting platform such as a Kubernetes cluster. Allow Dapr enabled applications to run on that platform and handles Dapr capabilities such as service discovery or sidecar injection. | [Self-hosted overview]({{< ref self-hosted-overview >}})<br />[Kubernetes overview]({{< ref kubernetes-overview >}})
+| Self-hosted | Your local Windows/macOS/Linux machine where you run your applications with Dapr. Dapr provides capabilities to run on your machine in "self-hosted" mode. | [Self-hosted mode]({{< ref self-hosted-overview.md >}})
```
Comment (msfussell) [inline daprdocs/content/en/concepts/terminology.md]: Self hosted is about running on machines/VMS where a single instance is a degenerate, special case. Especially now Consol is integrated with Dapr

**Review by msfussell — APPROVED** [review]
Did direct editing on files and left a few comments

File: daprdocs/content/en/concepts/overview.md (hunk 4 above)
Comment (AaronCrawfis) [reply daprdocs/content/en/concepts/overview.md]: I'll update

File: daprdocs/content/en/concepts/overview.md (hunk 5 above)
Comment (AaronCrawfis) [reply daprdocs/content/en/concepts/overview.md]: Idk, I'd still call the VSCode extension an "integration". We integrate Dapr building blocks, dubugging, and more into the VSCode experience. I'll update the header to say "Integrations and extensions"

**Pull request merged**